### PR TITLE
Reduce size of menu code

### PR DESCRIPTION
**Pull request by thinkyhead**
- Get rid of _selected functions, passing selected state instead
- Reduces code size by 6 - 7K !

### CODE REVIEW

File: Marlin/dogm_lcd_implementation.h [hunk 1]
```diff
@@ -282,7 +282,7 @@ static void lcd_implementation_status_screen() {
 }
 
 static void lcd_implementation_mark_as_selected(uint8_t row, char pr_char) {
-  if ((pr_char == '>') || (pr_char == LCD_STR_UPLEVEL[0] )) {
+  if ((pr_char == '>') || (pr_char == LCD_STR_UPLEVEL[0])) {
```
Comment (thinkyhead) [inline Marlin/dogm_lcd_implementation.h]: Actually...

  if (pr_char != ' ') { ... }

... should be fine here.

**Comment by AnHardt** [comment]
@thinkyhead 
In https://github.com/MarlinFirmware/Marlin/blob/Development/Marlin/dogm_lcd_implementation.h#L284
the value of the prechar is not used anymore since we use highlighting. You can save some more decisions by passing sel directly.

**Comment by thinkyhead** [comment]
@AnHardt Good idea! I guess I stuck to that because it was the convention, so it will be good to get past that and make the code more properly expressive.

**Comment by boelle** [comment]
this one seems logic to merge... but is it to early?

**Comment by thinkyhead** [comment]
With this last commit it should be ready. Waiting for Travis to pass it.

**Comment by AnHardt** [comment]
@thinkyhead 
There is a problem in 'ultralcd_implementation_hitachi_HD44780.h'.
While scroling down a menu all if fine until the line marker '>' reaches the ((bottom line) and (the end of the menu)).
When turning further the line marker vanishes.

May be the same as:
https://github.com/MarlinFirmware/Marlin/issues/1209#issuecomment-83794083

**Comment by AnHardt** [comment]
@thinkyhead 
'>' also not drawn for the 'edit' menu items.

**Comment by thinkyhead** [comment]
@AnHardt #1653 should fix the edit items. Not sure yet what's causing newly-scrolled items not to show as hilited.

**Comment by AnHardt** [comment]
@thinkyhead 
Yes, #1653 fixed the edit items.

No, not newly-scrolled items.
It makes the impression as if the '>' if scrolling out of view at the end of the list when you try to go beyond last item.

**Comment by AnHardt** [comment]
@thinkyhead 
This was really hard to find. Cost me hours. But at least i have a much better insight in the menu-code now.

**Comment by thinkyhead** [comment]
@AnHardt I see, so the menu was failing to scroll. Well, I'm excited you found the issue. Checking out the code now....

**Pull request merged**